pydantic_ai_slim/pydantic_ai/models/__init__.py

```diff
@@ -506,15 +509,49 @@ def cached_async_http_client(*, provider: str | None = None, timeout: int = 600,
 @cache
 def _cached_async_http_client(provider: str | None, timeout: int = 600, connect: int = 5) -> httpx.AsyncClient:
     return httpx.AsyncClient(
-        transport=_cached_async_http_transport(),
+        transport=_get_transport_for_loop(),
         timeout=httpx.Timeout(timeout=timeout, connect=connect),
         headers={'User-Agent': get_user_agent()},
     )
 
 
 @cache
-def _cached_async_http_transport() -> httpx.AsyncHTTPTransport:
-    return httpx.AsyncHTTPTransport()
+def _get_transport_for_loop() -> _PerLoopTransport:
+    return _PerLoopTransport()
```

Comment:
This is a convoluted way of just defining the instance as a module variable, but it works. Also, it was kinda-already done that way before.